numpy/core/src/umath/absolute.cpp

```diff
@@ -0,0 +1,88 @@
+#define _UMATHMODULE
+#define _MULTIARRAYMODULE
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+#undef HWY_TARGET_INCLUDE
+#define HWY_TARGET_INCLUDE "absolute.cpp"  // this file
+#include <hwy/foreach_target.h>  // must come before highway.h
+#include <hwy/highway.h>
+
+#include "numpy/ndarraytypes.h"
+#include "numpy/npy_common.h"
+#include "numpy/npy_math.h"
+#include "numpy/utils.h"
+
+namespace numpy {
+namespace HWY_NAMESPACE {  // required: unique per target
+
+// Can skip hn:: prefixes if already inside hwy::HWY_NAMESPACE.
+namespace hn = hwy::HWY_NAMESPACE;
+
+// Alternative to per-function HWY_ATTR: see HWY_BEFORE_NAMESPACE
+template <typename T>
+HWY_ATTR void SuperAbsolute(char **args, npy_intp const *dimensions, npy_intp const *steps) {
+  const T* HWY_RESTRICT input_array = (const T*) args[0];
+  T* HWY_RESTRICT output_array = (T*) args[1];
+  const size_t size = dimensions[0];
+  const hn::ScalableTag<T> d;
+  
+  for (size_t i = 0; i < size; i += hn::Lanes(d)) {
+    const auto in = hn::Load(d, input_array + i);
+    auto x = hn::Abs(in);
+    hn::Store(x, d, output_array + i);
+  }
+}
+
+HWY_ATTR void INT_SuperAbsolute(char **args, npy_intp const *dimensions, npy_intp const *steps) {
+  SuperAbsolute<npy_int>(args, dimensions, steps);
+}
+
+HWY_ATTR void DOUBLE_SuperAbsolute(char **args, npy_intp const *dimensions, npy_intp const *steps) {
+  SuperAbsolute<npy_double>(args, dimensions, steps);
+}
+
+HWY_ATTR void FLOAT_SuperAbsolute(char **args, npy_intp const *dimensions, npy_intp const *steps) {
+  SuperAbsolute<npy_float>(args, dimensions, steps);
+}
+
+}
+}
+
+#if HWY_ONCE
+namespace numpy {
+
+HWY_EXPORT(INT_SuperAbsolute);
+HWY_EXPORT(FLOAT_SuperAbsolute);
+HWY_EXPORT(DOUBLE_SuperAbsolute);
+
+extern "C" {
+
+NPY_NO_EXPORT void
+INT_absolute(char **args, npy_intp const *dimensions, npy_intp const *steps, void *NPY_UNUSED(func))
+{
+  static auto dispatcher = HWY_DYNAMIC_DISPATCH(INT_SuperAbsolute);
+  return dispatcher(args, dimensions, steps);
+}
+
+NPY_NO_EXPORT void
+DOUBLE_absolute(char **args, npy_intp const *dimensions, npy_intp const *steps, void *NPY_UNUSED(func))
+{
+  static auto dispatcher = HWY_DYNAMIC_DISPATCH(DOUBLE_SuperAbsolute);
+  return dispatcher(args, dimensions, steps);
+}
+
+NPY_NO_EXPORT void
+FLOAT_absolute(char **args, npy_intp const *dimensions, npy_intp const *steps, void *NPY_UNUSED(func))
+{
+  static auto dispatcher = HWY_DYNAMIC_DISPATCH(FLOAT_SuperAbsolute);
```

Comment:
```suggestion
  auto dispatcher = HWY_DYNAMIC_DISPATCH(FLOAT_SuperAbsolute);
```
Looks simple but I'm afraid of C++ evilness may involved since static initialization requires thread safety so this call usually will be warped in between a thread guard which make it actually slower than uses local variable if Highway cached CPUID calls similar to what NumPy does.
Here a pesudo code:
```Python
static the_dudced_type dispatcher;
static bool dispatcher_once = false
call cx_gaurd_lock
if not dispatcher_once
   dispatcher = HWY_DYNAMIC_DISPATCH(FLOAT_SuperAbsolute)
   dispatcher_once = true
endif
call cx_gaurd_unlock
```

Reply:
I'm guessing the even more ideal way to do this would be to generate the `HWY_DYNAMIC_DISPATCH` calls in the generated ufunc call.

```
// Generated code
sin_functions[4] = HWY_DYNAMIC_DISPATCH(FLOAT_SuperAbsolute);
```

Which could also generate the wrapper `FLOAT_SuperAbsolute` to avoid having to write it twice.